egs/librispeech/s5/local/nnet3/run_ivector_common.sh

```diff
@@ -0,0 +1,142 @@
+      utils/fix_data_dir.sh data/${datadir}_tmp
+
+      utils/copy_data_dir.sh --spk-prefix sp1.0- --utt-prefix sp1.0- data/${datadir} data/temp0
+      utils/combine_data.sh data/${datadir}_sp data/${datadir}_tmp data/temp0
+      utils/fix_data_dir.sh data/${datadir}_sp
+      rm -r data/temp0 data/${datadir}_tmp
+    done
+  fi
+
+  if [ $stage -le 2 ] && [ "$generate_alignments" == "true" ]; then
+    #obtain the alignment of the perturbed data
+    steps/align_fmllr.sh --nj 100 --cmd "$train_cmd" \
+      data/train_960_sp data/lang exp/tri6b exp/tri6b_sp || exit 1
+  fi
+  train_set=train_960_sp
+fi
+
+if [ $stage -le 3 ]; then
+  # Create high-resolution MFCC features (with 40 cepstra instead of 13).
+  # this shows how you can split across multiple file-systems.  we'll split the
+  # MFCC dir across multiple locations.  You might want to be careful here, if you
+  # have multiple copies of Kaldi checked out and run the same recipe, not to let
+  # them overwrite each other.
+  mfccdir=mfcc_hires
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $mfccdir/storage ]; then
+    utils/create_split_dir.pl /export/b0{1,2,3,4}/$USER/kaldi-data/egs/librispeech-$(date +'%m_%d_%H_%M')/s5/$mfccdir/storage $mfccdir/storage
+  fi
+
+  for datadir in $train_set test_clean test_other dev_clean dev_other; do
+    utils/copy_data_dir.sh data/$datadir data/${datadir}_hires
+    steps/make_mfcc.sh --nj 70 --mfcc-config conf/mfcc_hires.conf \
+      --cmd "$train_cmd" data/${datadir}_hires exp/make_hires/$datadir $mfccdir || exit 1;
+    steps/compute_cmvn_stats.sh data/${datadir}_hires exp/make_hires/$datadir $mfccdir || exit 1;
+  done
+
+  # now create some data subsets.
+  # mixed is the clean+other data.
+  # 30k is 1/10 of the data (around 100 hours), 60k is 1/5th of it (around 200 hours).
+  utils/subset_data_dir.sh data/${train_set}_hires 30000 data/${train_set}_mixed_hires_30k
+  utils/subset_data_dir.sh data/${train_set}_hires 60000 data/${train_set}_mixed_hires_60k
+fi
+
+if [ $stage -le 4 ]; then
+  # We need to build a small system just because we need the LDA+MLLT transform
+  # to train the diag-UBM on top of.  We align a subset of training data for
+  # this purpose.
+  utils/subset_data_dir.sh --utt-list <(awk '{print $1}' data/${train_set}_mixed_hires_30k/utt2spk) \
+     data/${train_set} data/${train_set}_30k
+
+  steps/align_fmllr.sh --nj 40 --cmd "$train_cmd" \
+    data/${train_set}_30k data/lang exp/tri6b exp/nnet3/tri6b_ali_30k
+fi
+
+if [ $stage -le 5 ]; then
+  # Train a small system just for its LDA+MLLT transform.  We use --num-iters 13
+  # because after we get the transform (12th iter is the last), any further
+  # training is pointless.
+  steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \
+    --realign-iters "" \
+    --splice-opts "--left-context=3 --right-context=3" \
+    5000 10000 data/${train_set}_mixed_hires_30k data/lang \
```

Comment:
I used 100k subset for this stage, the same as in the swbd. So the 
numbers should be enough.

Xingyu

在 2016/4/6 8:16, Yiming Wang 写道:
>
> In egs/librispeech/s5/local/nnet3/run_ivector_common.sh 
> <https://github.com/kaldi-asr/kaldi/pull/645#discussion_r58635511>:
>
> > +  # this purpose.
> > +  utils/subset_data_dir.sh --utt-list <(awk '{print $1}' data/${train_set}_mixed_hires_30k/utt2spk) \
> > +     data/${train_set} data/${train_set}_30k
> > +
> > +  steps/align_fmllr.sh --nj 40 --cmd "$train_cmd" \
> > +    data/${train_set}_30k data/lang exp/tri6b exp/nnet3/tri6b_ali_30k
> > +fi
> > +
> > +if [ $stage -le 5 ]; then
> > +  # Train a small system just for its LDA+MLLT transform.  We use --num-iters 13
> > +  # because after we get the transform (12th iter is the last), any further
> > +  # training is pointless.
> > +  steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \
> > +    --realign-iters "" \
> > +    --splice-opts "--left-context=3 --right-context=3" \
> > +    5000 10000 data/${train_set}_mixed_hires_30k data/lang \
> Hi Vijay, In Xingyu's fisher+swbd recipe, he used 5500 / 90000 for 
> #leaves / #gaussion, which is the same as the ones in swbd recipe. For 
> librispeech, do I need to halve these values (since the amount of data 
> in librispeech is ~half of the data in fisher_swbd) or some values 
> else? Yiming
> … <#>
> On Mon, Apr 4, 2016 at 4:39 PM, Vijayaditya Peddinti < 
> notifications@github.com> wrote: In 
> egs/librispeech/s5/local/nnet3/run_ivector_common.sh <#645 (comment) 
> <https://github.com/kaldi-asr/kaldi/pull/645#discussion_r58444580>>: > 
> + # this purpose. > + utils/subset_data_dir.sh --utt-list <(awk 
> '{print $1}' data/${train_set}_mixed_hires_30k/utt2spk) \ > + 
> data/${train_set} data/${train_set}_30k > + > + steps/align_fmllr.sh 
> --nj 40 --cmd "$train_cmd" \ > + data/${train_set}_30k data/lang 
> exp/tri6b exp/nnet3/tri6b_ali_30k > +fi > + > +if [ $stage -le 5 ]; 
> then > + # Train a small system just for its LDA+MLLT transform. We 
> use --num-iters 13 > + # because after we get the transform (12th iter 
> is the last), any further > + # training is pointless. > + 
> steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \ > + 
> --realign-iters "" \ > + --splice-opts "--left-context=3 
> --right-context=3" \ > + 5000 10000 data/${train_set}_mixed_hires_30k 
> data/lang \ Yes, I did verify that but I think we might have been 
> using smaller number of parameters in the old librispeech recipe. 
> Given that swbd recipe with 1/3rd the amount of data uses the same 
> number of parameters, I think you could probably increase these. Once 
> again communicate with @naxingyu <https://github.com/naxingyu> 
> <https://github.com/naxingyu> for the parameters he found helpful in 
> the 2100 hr task and interpolate the optimal parameter settings for 
> the 960hr librispeech task based on the 300hr swbd and 2100 
> fisher+swbd tasks. … <#m_-290847795370974225_> --Vijay On Mon, Apr 4, 
> 2016 at 4:29 PM, Yiming Wang <notifications@github.com> wrote: In 
> egs/librispeech/s5/local/nnet3/run_ivector_common.sh <#645 
> <https://github.com/kaldi-asr/kaldi/pull/645> (comment) <#645 
> (comment) 
> <https://github.com/kaldi-asr/kaldi/pull/645#discussion_r58442923>>>: 
> > + # this purpose. > + utils/subset_data_dir.sh --utt-list <(awk 
> '{print $1}' data/${train_set}_mixed_hires_30k/utt2spk) \ > + 
> data/${train_set} data/${train_set}_30k > + > + steps/align_fmllr.sh 
> --nj 40 --cmd "$train_cmd" \ > + data/${train_set}_30k data/lang 
> exp/tri6b exp/nnet3/tri6b_ali_30k > +fi > + > +if [ $stage -le 5 ]; 
> then > + # Train a small system just for its LDA+MLLT transform. We 
> use --num-iters 13 > + # because after we get the transform (12th iter 
> is the last), any further > + # training is pointless. > + 
> steps/train_lda_mllt.sh --cmd "$train_cmd" --num-iters 13 \ > + 
> --realign-iters "" \ > + --splice-opts "--left-context=3 
> --right-context=3" \ > + 5000 10000 data/${train_set}_mixed_hires_30k 
> data/lang \ These numbers are taken from 
> librispeech/s5/local/online/run_nnet2_common.sh. I noticed that 
> swbd/s5c/local/online/run_nnet2_common.sh and 
> swbd/s5c/local/nnet3/run_nnet3_common.sh also have the same numbers on 
> these. — You are receiving this because you were mentioned. Reply to 
> this email directly or view it on GitHub < 
> https://github.com/kaldi-asr/kaldi/pull/645/files/9ea0ff3e250cec5ed8f22b1a837c17791398eaf6#r58442923> 
> — You are receiving this because you were mentioned. Reply to this 
> email directly or view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/645/files/9ea0ff3e250cec5ed8f22b1a837c17791398eaf6#r58444580> 
>
> -- Yiming Wang Department of Computer Science The Johns Hopkins 
> University 3400 N. Charles St. Baltimore, MD 21218
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub 
> <https://github.com/kaldi-asr/kaldi/pull/645/files/9ea0ff3e250cec5ed8f22b1a837c17791398eaf6#r58635511>
>